pytorch_lightning/trainer/supporters.py

```diff
@@ -76,3 +76,17 @@ def _agg_memory(self, how: str):
                 return getattr(self.memory, how)()
             else:
                 return getattr(self.memory[:self.current_idx], how)()
+
+
+class Accumulator(object):
+    def __init__(self):
+        self.num_values = 0
+        self.total = 0
+
+    def accumulate(self, x):
+        with torch.no_grad():
+            self.total += x
+            self.num_values += 1
+
+    def mean(self):
```

Comment:
I'd rename this to `accumulate` and internally don't use a total value, but a list, that takes the tensors. That way, we could also use other accumulation types than just a mean (i.e. custom ones)